src/test/librbd/mirror/snapshot/test_mock_CreatePrimaryRequest.cc

```diff
@@ -181,12 +187,16 @@ class TestMockMirrorSnapshotCreatePrimaryRequest : public TestMockFixture {
                              boost::get<cls::rbd::MirrorSnapshotNamespace>(
                                &it->second.snap_namespace);
                            ASSERT_NE(nullptr, info);
-                           ASSERT_NE(0, info->mirror_peer_uuids.erase(
-                                       peer_uuid));
+                           if (!no_peer_test) {
+                             ASSERT_NE(0, info->mirror_peer_uuids.erase(
+                                         peer_uuids[count]));
+                           }
                            if (info->mirror_peer_uuids.empty()) {
                              mock_image_ctx.snap_info.erase(it);
                            }
                          }
+
+                         ++count;
```

Comment:
nit: I would like we avoid this static variable. What about making `peer_uuids` `std:list`, and alway check for the first element in the list (`peer_uuids.front()` instead of `peer_uuids[count]`), and then instead of `++count` remove the first element from the list (`peer_uuids.pop()`)?

Reply:
Good idea this is indeed cleaner.